launchable/commands/record/case_event.py

```diff
@@ -44,8 +44,13 @@ def f(case: TestCase, suite: TestSuite, report_file: str) -> TestPath:
     def from_case_and_suite(cls, path_builder: TestPathBuilder, case: TestCase, suite: TestSuite, report_file: str, data: Dict = None) -> Dict:
         "Builds a JSON representation of CaseEvent"
 
-        status = CaseEvent.TEST_FAILED if case.result and any(isinstance(case.result, s) for s in (
-            Failure, Error)) else CaseEvent.TEST_SKIPPED if isinstance(case.result, Skipped) else CaseEvent.TEST_PASSED
+        status = CaseEvent.TEST_PASSED
```

Comment:
[IMO] You can refactor this code more readable. For example,
```python
def result_type(case):
  if any(any(isinstance(r, s) for s in (Failure, Error)) for r in case.result):
    return CaseEvent.TEST_FAILED
  elif any(isinstance(r, Skipped) for r case.result):
    return CaseEvent.TEST_SKIPPED
  else:
    return CaseEvent.TEST_PASSED
```

Reply:
> I'm wondering that should we set success status to status's initial value.
> Will you please comment on it?

Does this code resolve your concern?

Reply:
I don't familiar with the Python way, but I want to avoid `if any(any(isinstance(r, s) for s in (Failure, Error)) for r in case.result):`

Because there are 2 loops and  `if` case into a line, it's difficult to understand the first time, I think.
What do you think about it?

Reply:
Yes, it wastes one more loop. But I think its performance does not matter in most cases. If we regard `for s in (Failure, Error)` as a constant, this is a `n + n` vs `n` problem, not `n^2`. At this time, you should choose a more readable and robust one. But if you think my idea is not good, please skip this. IMO less side effect is better in most cases.

Reply:
Sorry, I decided to keep this because I want to keep code and logic simple.
But, we have to keep consider the default value of status, so I'll add a comment about it and merge.